src/KubeOps/Operator/Builder/IComponentRegistrar.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Immutable;
+using k8s;
+using k8s.Models;
+
+namespace KubeOps.Operator.Builder
+{
+    internal interface IComponentRegistrar
+    {
+        public ImmutableHashSet<EntityRegistration> EntityRegistrations { get; }
+
+        public ImmutableHashSet<ControllerRegistration> ControllerRegistrations { get; }
+
+        public ImmutableHashSet<FinalizerRegistration> FinalizerRegistrations { get; }
+
+        public ImmutableHashSet<ValidatorRegistration> ValidatorRegistrations { get; }
+
+        public ImmutableHashSet<MutatorRegistration> MutatorRegistrations { get; }
+
+        IComponentRegistrar RegisterEntity<TEntity>()
+            where TEntity : IKubernetesObject<V1ObjectMeta>;
+
+        IComponentRegistrar RegisterController<TController, TEntity>()
+            where TController : class
```

Comment:
This may have been an artifact of me copying code around; needing to provide `TEntity` isn't an issue there since it is provided. I probably copied some of the method definitions from somewhere like `OperatorBuilderExtensions` where that would be an issue.